Sources/GRPCCodeGen/CodeGenerationRequest.swift

```diff
@@ -0,0 +1,224 @@
+    self.lookupDeserializer = lookupDeserializer
+  }
+
+  /// Represents an import: a module or a specific item from a module.
+  public struct Dependency {
+    /// If the dependency is an item, the property's value is the item representation.
+    /// If the dependency is a module, this property is nil.
+    public var item: Item? = nil
+
+    /// The name of the imported module or of the module an item is imported from.
+    public var module: String
+
+    public init(item: Item? = nil, module: String) {
+      self.item = item
+      self.module = module
+    }
+
+    /// Represents an item imported from a module.
+    public struct Item {
+      /// The keyword that specifies the item's kind (e.g.  `func`, `struct`).
+      public var kind: Kind
+
+      /// The imported item's symbol / name.
+      public var name: String
+
+      public init(kind: Kind, name: String) {
+        self.kind = kind
+        self.name = name
+      }
+      /// Represents the imported item's kind.
+      public struct Kind {
+        /// Describes the keyword associated with the imported item.
+        internal enum Value {
+          case `typealias`
+          case `struct`
+          case `class`
+          case `enum`
+          case `protocol`
+          case `let`
+          case `var`
+          case `func`
+        }
+
+        internal var value: Value
+
+        internal init(_ value: Value) {
+          self.value = value
+        }
+
+        /// The imported item is a typealias.
+        public static var `typealias`: Self {
+          Self(.`typealias`)
+        }
+
+        /// The imported item is a struct.
+        public static var `struct`: Self {
+          Self(.`struct`)
+        }
+
+        /// The imported item is a class.
+        public static var `class`: Self {
+          Self(.`class`)
+        }
+
+        /// The imported item is an enum.
+        public static var `enum`: Self {
+          Self(.`enum`)
+        }
+
+        /// The imported item is a protocol.
+        public static var `protocol`: Self {
+          Self(.`protocol`)
+        }
+
+        /// The imported item is a let.
+        public static var `let`: Self {
+          Self(.`let`)
+        }
+
+        /// The imported item is a var.
+        public static var `var`: Self {
+          Self(.`var`)
+        }
+
+        /// The imported item is a function.
+        public static var `func`: Self {
+          Self(.`func`)
+        }
+      }
+    }
+  }
+
+  /// Represents a service described in an IDL file.
+  public struct ServiceDescriptor {
+    /// Documentation from comments above the IDL service description.
+    public var documentation: String
+
+    /// Service name.
+    public var name: String
+
+    /// The service namespace.
+    ///
+    /// For `.proto` files it is the package name.
+    public var namespace: String
+
+    /// Array of descriptors for the methods of a service.
+    ///
+    /// - SeeAlso: ``MethodDescriptor``.
```

Comment:
Same comment as on `services`, this doesn't tell me anything more than `[MethodDescriptor]`.